rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -0,0 +1,151 @@
+---
+Number: 0006
+Category: Standards Track
+Status: Draft
+Author: Ke Wang
+Organization: Nervos Foundation
+Created: 2018-12-01
+---
+
+# Merkle Tree for Static Data
+
+## Complete Binary Merkle Tree
+
+CKB uses Complete Binary Merkle Tree(CBMT) to generate `Merkle Root` and `Merkle Proof` for a static list of items. Currently, CBMT is used to calculate `Transactions Root`. Basically, CBMT is a ***complete binary tree***, in which every level, except possibly the last, is completely filled, and all nodes are as far left as possible. And it is also a ***full binary tree***, in which every node other than the leaves has two children. Compare with other Merkle trees, the hash computation of CBMT is minimal, as well as the proof size.
+
+## Node Orginazation
+
+For the sake of illustration, we order the tree nodes from ***top to bottom*** and ***left to right***. In CBMT with `n` items, root is the `first` node, and the first item's hash is the `n-th` node, second is the `n+1-th` node, etc. We chose this orginazation because it is easy to caculate the node order for an item.
```

Comment:
I think it is a bit confused here to use `n-th` here. I recommend to revise it like this:

```
For the sake of illustration, we number the tree nodes from zero ***top to bottom*** and ***left to right***. In CBMT with `n` items, the root is the node *0*, and the first item's hash is the node *n*, the second is *n+1*, etc. We chose this organization because it is easy to calculate the node order for an item.
```